sources/packages/web/src/views/aest/institution/Programs.vue

```diff
@@ -1,13 +1,213 @@
+        :paginator="true"
+        :rows="defaultNoOfRows"
+        :rowsPerPageOptions="[10, 20, 50]"
+        :totalRecords="institutionProgramsSummary.programsCount"
+        @page="onPage($event)"
+        @sort="onSort($event)"
+        :loading="loading"
+      >
+        <Column field="submittedDate" header="Date Submitted" :sortable="true">
+          <template #body="slotProps">
+            <div class="p-text-capitalize">
+              {{ slotProps.data.formattedSubmittedDate }}
+            </div>
+          </template>
+        </Column>
+        <Column field="programName" header="Program Name">
+          <template #body="slotProps">
+            <div class="p-text-capitalize">
+              {{ slotProps.data.programName }}
+            </div>
+          </template>
+        </Column>
+        <Column field="locationName" header="Location">
+          <template #body="slotProps">
+            <div class="p-text-capitalize">
+              {{ slotProps.data.locationName }}
+            </div>
+          </template>
+        </Column>
+        <Column field="offeringsCount" header="Study periods">
+          <template #body="slotProps">
+            <div class="p-text-capitalize">
+              {{ slotProps.data.offeringsCount }}
+            </div>
+          </template>
+        </Column>
+        <Column field="programStatus" header="Status"
+          ><template #body="slotProps">
+            <Chip
+              :label="slotProps.data.programStatus"
+              class="p-mr-2 p-mb-2 p-text-uppercase"
+              :class="
+                getProgramStatusColorClass(slotProps.data.programStatus)
+              "/></template
+        ></Column>
+        <Column>
+          <template #body="slotProps">
+            <v-btn
+              outlined
+              @click="goToViewProgramDetail(slotProps.data.programId)"
+              >View</v-btn
+            >
+          </template>
+        </Column>
+      </DataTable>
+    </content-group>
   </content-group>
 </template>
 
 <script lang="ts">
+import { onMounted, ref, computed } from "vue";
+import { useRouter } from "vue-router";
+import { InstitutionService } from "@/services/InstitutionService";
 import ContentGroup from "@/components/generic/ContentGroup.vue";
+import {
+  AESTInstitutionProgramsSummaryPaginatedDto,
+  ApprovalStatus,
+} from "@/types";
+import { useToastMessage } from "@/composables";
+import { AESTRoutesConst } from "@/constants/routes/RouteConstants";
 export default {
   components: { ContentGroup },
-  //TODO, create and fetch an API to fetch Programs
+  props: {
+    institutionId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const toast = useToastMessage();
+    const router = useRouter();
+    const institutionProgramsSummary = ref(
+      {} as AESTInstitutionProgramsSummaryPaginatedDto,
+    );
+    const searchProgramName = ref("");
+    const defaultPage = 0;
+    const defaultNoOfRows = ref(10);
```

Comment:
Can we rename this to something like _pageSize_?
The "No" as an abbreviation for number I the midlle seems a bit confusing IMO.